src/Illuminate/Log/Context/ContextServiceProvider.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace Illuminate\Log\Context;
+
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Queue\Queue;
+use Illuminate\Support\Facades\Context;
+use Illuminate\Support\ServiceProvider;
+
+class ContextServiceProvider extends ServiceProvider
+{
+    /**
+     * Register the service provider.
+     *
+     * @return void
+     */
+    public function register()
+    {
+        $this->app->scoped(Repository::class);
+    }
+
+    /**
+     * Boot the application services.
+     *
+     * @return void
+     */
+    public function boot()
+    {
+        Queue::createPayloadUsing(function ($connection, $queue, $payload) {
+            $context = Context::dehydrate();
+
+            return $context === null ? $payload : [
+                ...$payload,
+                'illuminate:log:context' => $context,
+            ];
+        });
+
+        $this->app['events']->listen(function (JobProcessing $event) {
+            Context::hydrate($event->job->payload()['illuminate:log:context'] ?? null);
+        });
```

Comment:
hey @timacdonald, `JobRetryRequested` has a different interface for the payload.

for example, spatie/laravel-multitenacy has covered this [here](https://github.com/spatie/laravel-multitenancy/blob/main/src/Actions/MakeQueueTenantAwareAction.php#L88)
```php
protected function getEventPayload($event): ?array
    {
        return match (true) {
            $event instanceof JobProcessing => $event->job->payload(),
            $event instanceof JobRetryRequested => $event->payload(),
            default => null,
        };
    }
``` 

Am I missing something or this scenario was not mapped out?

Reply:
What issue are you seeing in particular?